src/routes/submission/index.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import SubmissionPage from './SubmissionPage';
+import { fetchSubmission } from '../../actions/submission';
+
+export default [
+  {
+    path: '/submission/:id',
+
+    async action({ params, store }) {
+      const submission = await store.dispatch(fetchSubmission(params.id));
+      if (!submission) {
+        return { redirect: '/' };
+      }
+      return {
+        title: `Submission: ${submission.summary}`,
+        component: <SubmissionPage submissionId={submission.id} />,
```

Comment:
Instead of feeding the object as props into the page it felt slightly more redux-y to connect the component and read from the store. What are your thoughts on that?